library/std/src/sys/pal/unix/net.rs

```diff
@@ -213,16 +213,25 @@ impl Socket {
                 }
                 0 => {}
                 _ => {
-                    // linux returns POLLOUT|POLLERR|POLLHUP for refused connections (!), so look
-                    // for POLLHUP rather than read readiness
-                    if pollfd.revents & libc::POLLHUP != 0 {
-                        let e = self.take_error()?.unwrap_or_else(|| {
-                            io::const_io_error!(
-                                io::ErrorKind::Uncategorized,
-                                "no error set after POLLHUP",
-                            )
-                        });
-                        return Err(e);
+                    if cfg!(target_os = "vxworks") {
+                        // Check if the connnection actually succeeded and return ok only when
+                        // the socket is ready and no errors were found
+                        // https://github.com/rust-lang/rust/issues/127018
+                        if let Some(e) = self.take_error()? {
+                            return Err(e);
+                        }
+                    } else {
+                        // linux returns POLLOUT|POLLERR|POLLHUP for refused connections (!), so look
+                        // for POLLHUP rather than read readiness
+                        if pollfd.revents & libc::POLLHUP != 0 {
```

Comment:
> I'm a bit surprised that no other targets reported an error on connect_timeout only checking POLLHUP in revents.

If you're up for it I think you can change this to check POLLERR too:

```suggestion
                        // linux returns POLLOUT|POLLERR|POLLHUP for refused connections (!), so check
                        // the error conditions.
                        if pollfd.revents & (libc::POLLHUP | libc::POLLERR) != 0 {
```

But definitely not required

Reply:
Sure. I'll set it to look for POLLHUP or POLLERR for linux.

Reply:
Thanks!